server/src/main/java/io/spine/server/integration/ThirdPartyContext.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.integration;
+
+import com.google.protobuf.Any;
+import io.spine.base.EventMessage;
+import io.spine.core.ActorContext;
+import io.spine.core.Event;
+import io.spine.core.EventContext;
+import io.spine.core.EventId;
+import io.spine.core.UserId;
+import io.spine.server.BoundedContext;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.base.Identifier.newUuid;
+import static io.spine.base.Time.currentTime;
+import static io.spine.protobuf.AnyPacker.pack;
+import static io.spine.server.BoundedContextBuilder.notStoringEvents;
+
+/**
+ * An external non-Spine based upstream system.
+ *
+ * <p>{@code ThirdPartyContext} helps to represent an external system as a Bounded Context. Events
+ * which occur in the external system are converted into domain events of the user's
+ * Bounded Contexts and dispatched via {@link IntegrationBus}.
+ *
+ * @implSpec Note that a {@code ThirdPartyContext} sends a request for external messages to
+ *         other contexts. The {@code ThirdPartyContext} never consumes external messages itself,
+ *         but requires the other Bounded Contexts to send their requests, so that the publishing
+ *         channels are open. Depending of the implementation of
+ *         {@link io.spine.server.transport.TransportFactory transport}, creating
+ *         a {@code ThirdPartyContext} may be an expensive operation. Thus, it is recommended that
+ *         the instances of this class are reused and {@linkplain #close() closed} when they are
+ *         no longer needed.
+ */
+public final class ThirdPartyContext implements AutoCloseable {
+
+    private final BoundedContext context;
+    private final Any producerId;
+
+    /**
+     * Creates a new single-tenant instance of {@code ThirdPartyContext} with the given name.
+     *
+     * @param name
+     *         name of the third-party system
```

Comment:
What we create is our representation of a _part_ of a 3rd party system, as if it was a Bounded Context. So, I'd say this: “A name of a Context which represents a part of a third-party system.“ Anyway, it should say that it's not the whole system.